File: bin/manta2alissa.pl
#!/usr/bin/env perl

use strict;
use Getopt::Long;

my $usage = qq{
perl my_script.pl
  Getting help:
    [--help]

  Input:
    [--infile filename]
		The name of the file to read. 
  Ouput:    
    [--outfile filename]
        The name of the output file. By default the output is the
        standard output
};

my $outfile = undef;
my $infile = undef;
my $help;

GetOptions(
    "help" => \$help,
    "infile=s" => \$infile,
    "outfile=s" => \$outfile);

# Print Help and exit
if ($help) {
    print $usage;
    exit(0);
}

if ($outfile) {
    open(STDOUT, ">$outfile") or die("Cannot open $outfile");
}

open (my $IN, "gunzip -c  $infile | ") or die "FATAL: Can't open file: $infile for reading.\n$!\n";

while (<$IN>) {

        my $line = $_;
        chomp $line;

	# The def line for the EnsEMBL accumulated score annotations
	if ($line =~ /^#.*/) {
		printf $line . "\n";
	} else {

		my @values = split(/\t/,$line);
                my ($chrom,$pos,$rsid,$ref,$alt,$qual,$filter,$info,$format) = @values[0..8];
		my @info = split( ";",$info);

		my $patients = "";
		if (scalar @values == 10) {
			$patients = @values[9];
		} else {
			$patients = join(@values[9..-1], "\t");
		}

		my $update = "";

		foreach my $i (@info) {

			next if ($i =~ /.*SVTYPE.*/) ;
			$update .= $i . ";" ;

		}	

		printf $chrom . "\t" . $pos . "\t" . $rsid . "\t" . $ref . "\t" . $alt . "\t" . $qual . "\t" . $filter . "\t" . $update . "\t" . $format . "\t"  . $patients . "\n";
	}
	
}

close($IN);

File: bin/manta2alissa.rb
#!/usr/bin/env ruby
# == NAME
# manta2alissa.rb
#
# == USAGE
# ./this_script.rb [ -h | --help ]
#[ -i | --infile ] |[ -o | --outfile ] | 
# == DESCRIPTION
# A skeleton script for Ruby
#
# == OPTIONS
# -h,--help Show help
# -i,--infile=INFILE input file
# -o,--outfile=OUTFILE : output file

#
# == EXPERT OPTIONS
#
# == AUTHOR
#  Marc Hoeppner, mphoeppner@gmail.com

require 'optparse'
require 'ostruct'

### Define modules and classes here

class VCFEntry

  attr_accessor :seq, :pos, :id, :ref, :alt, :qual, :filter, :info, :format, :samples

  def initialize(string)

    @seq, @pos, @id, @ref,@alt,@qual,@filter,@info,@format = string.strip.split("\t")[0..8]
    @samples = string.split("\t")[9..-1]

  end

  def to_s
	samples = self.samples.join("\t")
	return "#{self.seq}\t#{self.pos}\t#{self.id}\t#{self.ref}\t#{self.alt}\t#{self.qual}\t#{self.filter}\t#{self.info}\t#{self.format}\t#{samples}"
  end

end

### Get the script arguments and open relevant files
options = OpenStruct.new()
opts = OptionParser.new()
opts.banner = "A script description here"
opts.separator ""
opts.on("-i","--infile", "=INFILE","Input file") {|argument| options.infile = argument }
opts.on("-o","--outfile", "=OUTFILE","Output file") {|argument| options.outfile = argument }
opts.on("-h","--help","Display the usage information") {
 puts opts
 exit
}

opts.parse! 

options.infile ? input_stream = File.open(options.infile,"r") : input_stream = $stdin
options.outfile ? output_stream = File.new(options.outfile,'w') : output_stream = $stdout

while (line = input_stream.gets)

	line.strip!

	if line.match(/^#.*/)
		output_stream.puts line
	else

		entry = VCFEntry.new(line.strip)

		if entry.alt == "<INS>"
			entry.alt = "<DUP>"
		end
		if entry.id.include?("MantaINS")
			info = entry.info
			if entry.alt.match(/\<[A-Z]+\>/)
				entry.info = info.gsub(/SVTYPE\=[A-Z]+\;/, "SVTYPE=DUP;")
			else
	                        entry.info = info.gsub(/SVTYPE\=[A-Z]+\;/, "")
			end
		elsif entry.id.include?("MantaDEL") && entry.alt != "<DEL>"
                        info = entry.info
			entry.info = info.gsub(/SVTYPE\=[A-Z]+\;/, "")
		end
		output_stream.puts entry.to_s
	end

end

input_stream.close
output_stream.close
